File: backend/src/main/java/com/schoolfinder/config/AppProperties.java
package com.schoolfinder.config;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "app")
@Getter
@Setter
public class AppProperties {

    private Region region = new Region();
    private Firebase firebase = new Firebase();
    private Cloudinary cloudinary = new Cloudinary();
    private GoogleMaps googleMaps = new GoogleMaps();

    @Getter
    @Setter
    public static class Region {
        private String country = "Cameroon";
        private String currency = "XAF";
        // Centred on Yaoundé (the app lists universities in Yaoundé)
        private double mapCenterLat = 3.8480;
        private double mapCenterLng = 11.5021;
    }

    @Getter
    @Setter
    public static class Firebase {
        private boolean enabled = false;
        private String credentialsPath = "";
    }

    @Getter
    @Setter
    public static class Cloudinary {
        private String url = "";
    }

    @Getter
    @Setter
    public static class GoogleMaps {
        private String apiKey = "";
    }
}

File: backend/src/main/kotlin/com/schoolfinder/config/AppProperties.kt
package com.schoolfinder.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "app")
data class AppProperties(
    val region: Region = Region(),
    val firebase: Firebase = Firebase(),
    val cloudinary: Cloudinary = Cloudinary(),
    val googleMaps: GoogleMaps = GoogleMaps(),
) {
    data class Region(
        val country: String = "Cameroon",
        val currency: String = "XAF",
        val mapCenterLat: Double = 4.0511,
        val mapCenterLng: Double = 9.7679,
    )

    data class Firebase(
        val enabled: Boolean = false,
        val credentialsPath: String = "",
    )

    data class Cloudinary(
        val url: String = "",
    )

    data class GoogleMaps(
        val apiKey: String = "",
    )
}
